src/components/DataProvider/index.js

```diff
@@ -492,8 +492,16 @@ export default class DataProvider extends Component {
         [id]: { ...loaderConfig },
       };
     }
+
     this.setState(stateUpdates, () => {
-      this.props.onFetchData();
+      const lastPoints = {};
+      if (stateUpdates.contextSeries) {
```

Comment:
Shouldn't you be iterating on `this.getSeriesObjects()` or something? `contextSeries` are series which will be drawn in the context chart (we should clean this up at some point ...)